packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/common_use_cases/fetch_access_token_use_case.dart

```diff
@@ -0,0 +1,24 @@
+import '../repositories/auth_repository.dart';
+
+class FetchAccessTokenUseCase {
+  FetchAccessTokenUseCase(
+      this._authRepository,
+      );
+
+  final AuthRepository _authRepository;
+
+  Future<String?> execute({bool forceFetchNewToken= false}) async {
```

Comment:
Should I change this with (https://pub.dev/packages/shared_preferences) instead it?

Reply:
Ideally it would be good if we could use both of them. Depending on the project needs we may not want to have a web app and in that case the flutter secure storage is a good idea. On the web however, we can not use the secure storage (since there is no support for it, as of today). However, we can have another local storage that implements shared preferences which do support mobile and web.

It would be good if we had both implementations and if we could switch them out easily, according to our needs. We could treat the implementation with shared preferences as the default one.

Reply:
For this, maybe define a contract that the implementations would use. This way we can easily swap them out. We could even add our own implementations too.